webui/react/src/pages/ModelRegistry.tsx

```diff
@@ -682,15 +679,19 @@ const ModelRegistry: React.FC<Props> = ({ workspace }: Props) => {
       children: React.ReactNode;
       onVisibleChange?: (visible: boolean) => void;
       record: ModelItem;
-    }) => (
-      <Dropdown
-        menu={ModelActionMenu(record)}
-        trigger={['contextMenu']}
-        onOpenChange={onVisibleChange}>
-        {children}
-      </Dropdown>
-    ),
-    [ModelActionMenu],
+    }) => {
+      const canDelete = canDeleteModel({ model: record });
+      const canModify = canModifyModel({ model: record });
+      return canDelete || canModify ? (
+        <Dropdown
+          menu={ModelActionMenu(record, canDelete, canModify)}
+          trigger={['contextMenu']}
+          onOpenChange={onVisibleChange}>
+          {children}
+        </Dropdown>
+      ) : null;
+    },
+    [ModelActionMenu, canDeleteModel, canModifyModel],
```

Comment:
this causes the issue we talked yesterday.
Even though user has `viewer` and `modelRegistryViewer` permissions, user cannot see the models

Reply:
how is the dropdown causing the models to not be visible?

Reply:
i didnt dive into it much, but it was working when i reverted the code. 
prob its interactive table logic issue?? https://github.com/determined-ai/determined/blob/master/webui/react/src/components/Table/InteractiveTable.tsx#L191

Reply:
ah i think we shouldn't return null.

Reply:
ok I can prevent this from returning null.

Reply:
it seems working but need double check

```diff
diff --git a/webui/react/src/pages/ModelRegistry.tsx b/webui/react/src/pages/ModelRegistry.tsx
index 182e8cd99..d18080a6f 100644
--- a/webui/react/src/pages/ModelRegistry.tsx
+++ b/webui/react/src/pages/ModelRegistry.tsx
@@ -421,16 +421,14 @@ const ModelRegistry: React.FC<Props> = ({ workspace }: Props) => {
         funcs[e.key as ValueOf<typeof MenuKey>]();
       };
 
-      const menuItems: MenuProps['items'] = [];
-      if (canModify) {
-        menuItems.push({
+      const menuItems: MenuProps['items'] = [
+        {
+          disabled: !canModify,
           key: MenuKey.SwitchArchived,
           label: record.archived ? 'Unarchive' : 'Archive',
-        });
-        if (!record.archived) {
-          menuItems.push({ key: MenuKey.MoveModel, label: 'Move' });
-        }
-      }
+        },
+        { disabled: !canModify || record.archived, key: MenuKey.MoveModel, label: 'Move' },
+      ];
       if (canDelete) {
         menuItems.push({ danger: true, key: MenuKey.DeleteModel, label: 'Delete Model' });
       }
@@ -682,14 +680,14 @@ const ModelRegistry: React.FC<Props> = ({ workspace }: Props) => {
     }) => {
       const canDelete = canDeleteModel({ model: record });
       const canModify = canModifyModel({ model: record });
-      return canDelete || canModify ? (
+      return (
         <Dropdown
           menu={ModelActionMenu(record, canDelete, canModify)}
           trigger={['contextMenu']}
           onOpenChange={onVisibleChange}>
           {children}
         </Dropdown>
-      ) : null;
+      );
     },
     [ModelActionMenu, canDeleteModel, canModifyModel],
   );
```